src/fulfillment/README.md

```diff
@@ -0,0 +1,74 @@
+# Fulfillment
+---
+This service represents a fulfillment system in an e-commerce platform. 
+The key idea demonstrated by this service is asynchronous, message based application design.
+This allows us to scale the order management and fulfillment systems independently and provides 
+for better fault tolerance. A messaging system also makes it easy for programs to communicate 
+across different environments, languages, cloud providers and on-premise systems. Generally more 
+throughput can be achieved with these designs than traditional synchronous, blocking models.
+
+## Building
+
+The application uses a `gradle` build. However for convenience, it can be build inside a 
+docker container. To build the application execute the following command (from the repository root)
+
+```shell script
+docker build -t mushop-fulfillment:latest src/fulfillment
+``` 
+This is a multi stage that sets-up the the build environment and generates a docker image 
+that runs the application. By default, the application that is built by `gradle` is processed by the 
+GraalVM native image builder that compiles the application in to a native binary.
+
+
+> **_NOTE:_** GraalVM's native image build is cpu and memory intensive, and could take a few minutes to complete.
+> Make sure your docker daemon has been allocated enough CPUs and memory.
+
+### Building a JVM application
+
+If you would like to build the application as a standard executable jar file, we have 
+provided an alternate `Dockerfile` that will skip the GraalVM based native image generation 
+and build an image that uses a standard JVM to run the application. This is useful for either comparing 
+GraalVM against the normal JVM execution, as well as for rapid development.
+
+```shell script
+docker build -t mushop-fulfillment:latest -f src/fulfillment/Dockerfile.jvm src/fulfillment
+``` 
+
+## Running 
+
+The fulfillment service is included in the `docker-compose` configuration can simply be run using `docker-compose`.
+To run the container stand-alone, simply run the following command.
+
+```shell script
+docker run -p 8099:8099  mushop-fulfillment:local
+``` 
+
+This runs the container locally and exposes the application's api to the docker host's port `8099`.
+You can validate the application state by simply making a health check request to 
+
+```shell script
+curl localhost:8099/health
```

Comment:
Why using 8099 port?

Reply:
App happens to listen on port 8099. I can update the docker run command to map it to something else.

Reply:
These other services are not using 80 because they run as non root. Obviously, the Service definitions map 80:xxxx

Reply:
changed to 80.